server/handler.go

```diff
@@ -0,0 +1,109 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/benchttp/engine/internal/configparse"
+	"github.com/benchttp/engine/internal/socketio"
+	"github.com/benchttp/engine/runner"
+	"github.com/gorilla/websocket"
+)
+
+// Handler has as single method, Handler.ServeHTTP.
+// It serves a websocket server allowing remote
+// manipulation of runner.Runner.
+type Handler struct {
+	Silent   bool
+	Token    string
+	srv      *service
+	upgrader websocket.Upgrader
+}
+
+func NewHandler(silent bool, token string) *Handler {
+	return &Handler{
+		Silent: silent,
+		Token:  token,
+		srv:    &service{},
+		upgrader: websocket.Upgrader{
+			CheckOrigin: func(r *http.Request) bool {
+				return r.URL.Query().Get("access_token") == token
+			},
+		},
+	}
+}
+
+func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	switch r.URL.Path {
+	case "/run":
+		h.handle(w, r)
+	default:
+		http.NotFound(w, r)
+	}
+}
+
+func (h *Handler) handle(w http.ResponseWriter, r *http.Request) {
+	ws, err := h.upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		log.Println(err)
+		return
+	}
+
+	defer ws.Close()
+	// The client is gone, flush all the state.
+	// TODO Handle reconnect?
+	defer h.srv.flush()
```

Comment:
We need to reverse the two statements if we want to make sure the client is closed first:
```suggestion
	defer h.srv.flush()
	// The client is gone, flush all the state.
	// TODO Handle reconnect?
	defer ws.Close()
```

We could probably make it less error prone using a single `defer`:
```suggestion
	defer func() {
		ws.Close()
		h.srv.flush()
	}
```